lib/event_tracer/prometheus_logger.rb

```diff
@@ -0,0 +1,99 @@
+module EventTracer
+  class PrometheusLogger < BasicDecorator
+
+    SUPPORTED_METRIC_TYPES = {
+      counter: :increment_count,
+      gauge: :set_gauge
+    }.freeze
+    DEFAULT_INCREMENT = 1
+
+    attr_reader :allowed_tags
+
+    def initialize(prometheus, allowed_tags: [], default_tags: {}, raise_if_missing: true)
+      super(prometheus)
```

Comment:
We can pass either `Prometheus::Client.registry` for app process and `SidekiqPrometheus.registry` for Sidekiq process. 
They both have the same interface to register metrics, and increment, set metrics